Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -78,60 +82,54 @@ public class NotificationPublisher : INotificationPublisher {
       return new[] { first, second }.Min();
     }
 
-    private void PublishDocumentDiagnostics(IdeState previousState, IdeState state) {
-      var previousStateDiagnostics = previousState.GetDiagnostics();
-      var currentDiagnostics = state.GetDiagnostics();
-      if (state.Compilation.Project.IsImplicitProject) {
-        PublishImplicitProjectDiagnostics(state, currentDiagnostics, previousStateDiagnostics);
-      } else {
-        var uris = previousStateDiagnostics.Keys.Concat(currentDiagnostics.Keys).Distinct();
-
-        foreach (var uri in uris) {
-          IEnumerable<Diagnostic> current = currentDiagnostics.GetOrDefault(uri, Enumerable.Empty<Diagnostic>);
-          IEnumerable<Diagnostic> previous = previousStateDiagnostics.GetOrDefault(uri, Enumerable.Empty<Diagnostic>);
-          if (previous.SequenceEqual(current)) {
-            continue;
-          }
+    private Dictionary<Uri, IList<Diagnostic>> publishedDiagnostics = new();
 
-          languageServer.TextDocument.PublishDiagnostics(new PublishDiagnosticsParams {
-            Uri = uri,
-            Version = filesystem.GetVersion(uri),
-            Diagnostics = current.ToList(),
-          });
-        }
-      }
-    }
+    private async Task PublishDocumentDiagnostics(IdeState state) {
```

Comment:
Not a blocker, just a question - was the branch on `state.Compilation.Project.IsImplicitProject` removed because projects created by `ProjectManagerDatabase.ImplicitProject()` "just work" as expected?

Reply:
The code works for both `ImplicitProject` and `!ImplicitProject`